### ISSUE TITLE
UnifiedMap: Same waypoints being added/visible multiple times to/on the map

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#15595]
Having been on a short tour today I noticed a strange behavior wrt to waypoints as part of an individual route:

Somehow I managed to have the same waypoint twice or even three times on the map, so when long-tapping on that waypoint (to get the long-tap popup to remove it from the individual route), that waypoint got listed twice or even three times in the popup menu.

IMHO the underlaying data structures should prevent adding a waypoint (or a cache) multiple times to its list. On adding the same element (identified by its GC code / waypoint id), the existing version should be removed first.

**Comment by moving-bits** [comment]
May be the same or have the same cause as #15412 (haven't checked the displayed route length at that moment)

**Comment by ztNFny** [comment]
Yes, this is "standard" behaviour with routing on unified map. it's pretty much unusable - which is the main reason i still use old maps for trails. I considered it covered by https://github.com/cgeo/cgeo/issues/15412

**Comment by moving-bits** [comment]
Does it still happen? I haven't experienced that behavior since back then when I opened this issue.

**Comment by ztNFny** [comment]
Happens all the time here. Routing on UM is pretty much unusable because of it.

**Comment by moving-bits** [comment]
This is really strange. Seems that some devices are in sort of a constant conflict with UnifiedMap, whereas others are not. I'd like to know what (or if) the conflicting devices have in common (either hardware, software or c:geo settings).

(I'm using routing with UM all the time, last time was yesterday with current beta version, normally without any problems at all.)

**Comment by moving-bits** [comment]
From today's dev meeting: Hasn't happened for me since a long time, so let's close here for now.
